osbuild/util/ostree.py

```diff
@@ -215,15 +214,52 @@ def parse_input_commits(commits):
 def deployment_path(root: PathLike, osname: str, ref: str, serial: int):
     """Return the path to a deployment given the parameters"""
 
-    base = os.path.join(root, "ostree")
+    if osname == "" and ref == "" and serial == None:
+        filenames =  glob.glob(root + '/ostree/deploy/*/deploy/*.0', recursive=True)
+        if len(filenames) < 1:
+            raise ValueError("Cound not find deployment")
+        elif len(filenames) > 1:
+            raise ValueError("More than one deployment found")
+        return filenames[0]
+    
+    else:
+        base = os.path.join(root, "ostree")
+
+        repo = os.path.join(base, "repo")
+        stateroot = os.path.join(base, "deploy", osname)
 
-    repo = os.path.join(base, "repo")
-    stateroot = os.path.join(base, "deploy", osname)
+        commit = rev_parse(repo, ref)
+        sysroot = f"{stateroot}/deploy/{commit}.{serial}"
 
-    commit = rev_parse(repo, ref)
-    sysroot = f"{stateroot}/deploy/{commit}.{serial}"
+        return sysroot
+    
 
-    return sysroot
+def parse_origin(origin: PathLike):
+    """Parse the origin file and return the deployment type and imgref
+
+    Example container case: container-image-reference=ostree-remote-image:fedora:docker://quay.io/fedora/fedora-coreos:stable
+    Example ostree commit case: refspec=fedora:fedora/x86_64/coreos/stable
+    """
```

Comment:
@jlebon I moved the code that parses the origin file into `ostree.py` such that other stages outside the aleph stage can utilize it. The functions takes in the path to the origin file as a parameter, but do you think auto detecting the path to the origin file should also be embedded in this function?

Reply:
No strong opinion, though I would just leave that functionality out and let callers call out to `deployment_path()` themselves.